src/lib.rs

```diff
@@ -2168,6 +2187,18 @@ impl Build {
         };
 
         let (sdk_prefix, sim_prefix, min_version) = match os {
+            Os::MacOs => (
+                "macosx",
+                "",
+                std::env::var("MACOSX_DEPLOYMENT_TARGET").unwrap_or_else(|_| {
+                    (if arch_str == "aarch64" {
+                        "11.0"
+                    } else {
+                        "10.7"
+                    })
+                    .into()
+                }),
```

Comment:
The `rustc` PR for `--print deployment-target` just landed (https://github.com/rust-lang/rust/pull/105354), and should be available starting in 1.71 fyi. Ideally that would be queried for the version and then falling back to this hardcoded logic if the compiler is too old.